lib/RequestRouter.js

```diff
@@ -0,0 +1,809 @@
+};
+
+RequestRouter.prototype.getProps = function() {
+    return this._request.getProps();
+};
+
+RequestRouter.prototype.setProps = function(props) {
+    this._request.setProps(props);
+    return this;
+};
+
+RequestRouter.prototype.setMethod = function(method) {
+    this._request.setMethod(method);
+    return this;
+};
+
+RequestRouter.prototype.getMethod = function() {
+    return this._request.getMethod();
+};
+
+RequestRouter.prototype.isOptions = function() {
+    return this._request.isOptions();
+};
+
+RequestRouter.prototype.isGet = function() {
+    return this._request.isGet();
+};
+
+RequestRouter.prototype.isPost = function() {
+    return this._request.isPost();
+};
+
+RequestRouter.prototype.setParams = function(params) {
+    this._request.setParams(params);
+
+    //Validate params
+    var params = validateAndCast.call(this, params);
+    if (params) {
+        this._request.setParams(params);
+    }
+    return this;
+};
+
+RequestRouter.prototype.getParams = function() {
+    return this._request.getParams();
+};
+
+RequestRouter.prototype.setParam = function(key, value) {
+    this._request.setParam(key, value);
+    var returnParam = validateAndCast.call(this, {key : value});
+
+    if (returnParam && returnParam.hasOwnProperty(key)) {
+        this._request.setParam(key, returnParam[key]);
+    }
+
+    return this;
+};
+
+RequestRouter.prototype.getParam = function(param) {
+    return this._request.getParam(param);
+};
+
+RequestRouter.prototype.setQueries = function(queries) {
+    this._request.setQueries(queries);
+    //Validate queries
+    var queries = validateAndCast.call(this, queries);
+    if (queries) {
+        this._request.setQueries(queries);
+    }
+    return this;
+};
+
+RequestRouter.prototype.getQueries = function() {
+    return this._request.getQueries();
+};
+
+RequestRouter.prototype.setQuery = function(key, value) {
+    this._request.setQuery(key, value);
+    var returnQuery = validateAndCast.call(this, {key : value});
+
+    if (returnQuery && returnQuery.hasOwnProperty(key)) {
+        this._request.setQuery(key, returnQuery[key]);
+    }
+    return this;
+};
+
+RequestRouter.prototype.unsetQuery = function(query) {
+    this._request.unsetQuery(query);
+    return this;
+};
+
+RequestRouter.prototype.getQuery = function(query) {
+    return this._request.getQuery(query);
+};
+
+function cleanParamKey(key) {
+    if (key.toString().substr(-2) == '[]' || key.toString().substr(-2) == '{}') {
+        return key.toString().slice(0, key.length - 2);
+    }
+}
+
+function castParam(key, value, type) {
+    var hasTypeError = true;
+    if (type.indexOf('boolean') !== -1) {
+        switch(value) {
+            case 'true' :
+            case true :
+                return true;
+            case 'false':
+            case false:
+                return false;
+            default:
+                hasTypeError = true;
+        }
+    }
+
+    if (type.indexOf('integer') !== - 1) {
```

Comment:
it's this unit test:

does not throw error if param is a float with type int and float in router-test.js

So what's gonna happen is that it's not even gonna attempt to parseInt because 1.23 don't validate against the regex... then it'll go in the check of type.indexOf('float') and it'll parseFloat()

Is that the use case you meant?